services/delegation/src/modules/blockchain/blockchain.service.ts

```diff
@@ -110,6 +124,43 @@ export class BlockchainService {
     return false;
   }
 
+  async undelegate(address: string, amount: ethers.BigNumber) {
+    try {
+      const tx = await this.contract.undelegate(address, {
+        gasPrice: this.provider.getGasPrice(),
+        value: amount,
+      });
+      await tx.wait();
+      if (tx) {
+        const receipt = await this.provider.getTransactionReceipt(tx.hash);
+        return receipt.blockNumber;
+      }
+    } catch (e) {
+      console.error(
+        `Could not undelegate ${amount.toString()} TARA to validator ${address}`,
+        e,
+      );
+    }
+  }
+
+  async confirmUndelegate(address: string) {
+    try {
+      const tx = await this.contract.undelegate(address, {
```

Comment:
We wanna call confirm, not undelegate again.

Reply:
Right, corrected.